Samples/HelloWorld/NonSiloTests/TestGrainFactory.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Moq;
+using Orleans;
+using Orleans.Core;
+using Orleans.Runtime;
+using Orleans.Storage;
+
+namespace NonSiloTests
+{
+    public static class TestGrainFactory
+    {
+        private static readonly GrainCreator GrainCreator;
+
+        private static readonly IStorageProvider StorageProvider;
+
+        static TestGrainFactory()
+        {
+            var grainRuntime = Mock.Of<IGrainRuntime>();
+
+            var mockStorageProvider = new Mock<IStorageProvider>();
+
+            StorageProvider = mockStorageProvider.Object;
+
+            GrainCreator = new GrainCreator(grainRuntime);
```

Comment:
I figured showing an example of how I was using GrainCreator would help to clarify some of the confusion. I think the #1918 should make the API much cleaner for these types of tests. Thanks for the feedback.